contrib/go-chi/chi/chi.go

```diff
@@ -0,0 +1,45 @@
+// Package chi provides tracing functions for tracing the go-chi/chi package (https://github.com/go-chi/chi).
+package chi // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/go-chi/chi"
+
+import (
+	"net/http"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/internal/httputil"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+
+	"github.com/go-chi/chi"
+)
+
+// Router registers routes to be matched and dispatches a handler.
+type Router struct {
+	chi.Router
+	config *routerConfig
+}
+
+// NewRouter returns a new router instance traced with the global tracer.
+func NewRouter(opts ...RouterOption) *Router {
+	cfg := new(routerConfig)
+	defaults(cfg)
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	return &Router{
+		Router: chi.NewRouter(),
+		config: cfg,
+	}
+}
+
+// ServeHTTP dispatches the request to the handler
+// whose pattern most closely matches the request URL.
+// We only need to rewrite this function to be able to trace
+// all the incoming requests to the underlying multiplexer
+func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
```

Comment:
Please rewrite this function as follows:
```go
func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
    resource := req.Method + " " + req.URL.Path
    httputil.TranceAndServe(r.Router, w, req, r.config.ServiceName, resource, r.config.spanOpts...)
}
```
It should suffice.

Reply:
Made the changes!! Thanks!!

Reply:
Was a spelling error with `TranceAndServe`. Should be `TraceAndServe` instead.